### PR TITLE
Only create one Service with one EndpointSlice per every 100 compute …

### PR DESCRIPTION
**Pull request by jlarriba**
…nodes to avoid overloading the control plane

### CODE REVIEW

**Comment by slagle** [comment]
recheck

File: pkg/deployment/kube_service.go [hunk 1]
```diff
@@ -38,39 +38,53 @@ func CreateKubeServices(
...
+			return err
+		}
+
+		addresses := make([]string, len(nodes.Items))
+		for i, item := range nodes.Items {
+			addresses[i] = item.Spec.AnsibleHost
+		}
+
+		index := 0
+		for i := 0; i < len(addresses); i += 100 {
```
Comment (slagle) [inline pkg/deployment/kube_service.go:52]: should the 100 be configurable as a field on DataPlaneService?

Comment (jlarriba) [reply pkg/deployment/kube_service.go:52]: No, 100 is a hard limit imposed on EndpointSlices by Kubernetes itself, it is not an arbitrary limit. In a "normal" Service, Kubernetes would spawn a new EndpointSlice every 100 endpoints, as described in documentation:

"By default, Kubernetes makes a new EndpointSlice once the existing EndpointSlices all contain at least 100 endpoints. Kubernetes does not make the new EndpointSlice until an extra endpoint needs to be added."

Here, we are mimicking that behaviour as Compute nodes are not Kubernetes objects.

**Comment by jlarriba** [comment]
recheck

File: pkg/deployment/kube_service.go (hunk 1 above)
Comment (slagle) [reply pkg/deployment/kube_service.go:52]: :+1:

**Review by slagle — COMMENTED** [review]
+2

**Comment by jlarriba** [comment]
> +2

Could you please approve and lgtm?

**Review by abays — APPROVED** [review]
/lgtm

**Comment by abays** [comment]
> As Zuul has passed all the jobs independently, someone can merge this manually?

Has the baremetal job passed?  I don't see a successful execution.

**Comment by jlarriba** [comment]
True, I was wrong.

I have been reviewing the logs and everything seems to be working perfectly: the operator and the ansibleee executions. It seems to be a permissions issue from the CI: 

`"E0809 08:25:54.138639   23810 reflector.go:140] k8s.io/client-go/tools/watch/informerwatcher.go:146: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: Unauthorized", "error: timed out waiting for the condition on openstackdataplanes/openstack-edpm"]`

So, lets see with another recheck.

**Comment by jlarriba** [comment]
recheck

**Comment by abays** [comment]
recheck

**Comment by abays** [comment]
recheck

**Comment by abays** [comment]
recheck

**Comment by abays** [comment]
recheck

**Review by slagle — APPROVED** [review]

**Pull request merged**